lib/dns.js

```diff
@@ -213,6 +215,11 @@ function resolver(bindingName) {
       throw new ERR_INVALID_CALLBACK();
     }
 
+    try {
```

Comment:
Would this be needed if we were only using the ICU based `toASCII`?

Reply:
My guess is no